packages/pds/src/stream/types.ts

```diff
@@ -0,0 +1,25 @@
+import Database from '../db'
+
+export type MessageOfType<T extends string = string> = {
+  type: T
+  [s: string]: unknown
+}
+
+export type Listener<M extends MessageOfType = MessageOfType> = (ctx: {
+  db: Database
+  message: M
+}) => Promise<void | MessageOfType[]>
+
+export type Listenable<M extends MessageOfType = MessageOfType> =
```

Comment:
i see you went to some extra ends to differentiate Listeners & Consumers, but it looks like we're only using Consumers at this point. Just out of curiosity, what do you have in mind for non-consuming listeners?

Reply:
& kinda wondering if we can simplify some of the `Listenable` song & dance (checking two locations, etc). but not overly worried about it, cause I can tell you have a plan w it 👌

Reply:
TBH I was a little bit in library-writing mindset, and some of this was about conveniences that we don't actually need!  To sort of split the difference I am going to keep the `Listenable` interface separate from the `Consumer` class, but nix the ability to pass a listener callback as a listenable.  That should be a little simpler :)